chainntnfs/interface.go

```diff
@@ -808,3 +808,12 @@ type ConfirmHintCache interface {
 	// the cache.
 	PurgeConfirmHint(confRequests ...ConfRequest) error
 }
+
+// MempoolWatcher defines an interface that allows the caller to query
+// information in the mempool.
+type MempoolWatcher interface {
```

Comment:
:+1 see other comment about moving to a diff pacakge/module.

Reply:
this is now changed as the `ChainNotifier` has pretty much implemented everything already, and we only need to make sure to subscribe the inputs and read the notifications sent back. Meanwhile, the mempool notifier mimicks the design of `chainntnfs.TxNotifier`.

Meanwhile feel like the whole notifier can be moved into `btcwallet`...another topic another pr I guess

Reply:
> Meanwhile feel like the whole notifier can be moved into btcwallet...another topic another pr I guess

One thing I've been meaning to do, now that we have generics, is refactor a lot of the notifier implementations themselves. Right now we have 3 pretty much direct copies of the same code with slight tweaks. We should instead made a `ChainFeeder` or w/e interface, then make a single `ChainNotifier[T ChainFeeder]` struct. This should cut down on a lot of code, and also make it easier to ensure changes are consistent across all the backends we support.